### PR TITLE
`BlueprintViewMetricsDelegate` terminology and calculation improvements

### PR DESCRIPTION
**Pull request by robmaceachern**
This renames a few items related to the `BlueprintViewMetricsDelegate` and improves the calculation of the metric values.

- Renamed `BlueprintViewUpdateMetrics` to `BlueprintViewRenderMetrics`.
- Renamed `BlueprintViewRenderMetrics.measureDuration` to `layoutDuration`.
- Renamed `BlueprintViewMetricsDelegate.blueprintView(_:completedUpdateWith:)` to `blueprintView(_:completedRenderWith:)`.
- `BlueprintViewRenderMetrics` values are now calculated using `CACurrentMediaTime` instead of `Date`.
- Tests for `BlueprintViewMetricsDelegate`.

### CODE REVIEW

File: BlueprintUI/Tests/BlueprintViewTests.swift [hunk 1]
```diff
@@ -691,6 +691,26 @@ class BlueprintViewTests: XCTestCase {
...
+
+        guard let metric = testMetricsDelegate.metrics.first else {
+            XCTFail("No metric generated")
+            return
+        }
+
+        XCTAssertTrue(metric.totalDuration > 0)
+        XCTAssertTrue(metric.layoutDuration > 0)
+        XCTAssertTrue(metric.viewUpdateDuration > 0)
+        XCTAssertEqual(metric.totalDuration, metric.layoutDuration + metric.viewUpdateDuration, accuracy: .ulpOfOne)
```
Comment (robmaceachern) [inline BlueprintUI/Tests/BlueprintViewTests.swift:712]: I'm no `ulp` expert and I was debating doing this instead:

```
XCTAssertEqual(metric.totalDuration, metric.layoutDuration + metric.viewUpdateDuration, accuracy: metric.totalDuration.ulp)
```

File: BlueprintUI/Sources/BlueprintView/BlueprintView.swift [hunk 2]
```diff
@@ -375,18 +375,18 @@ public final class BlueprintView: UIView {
         }
 
         Logger.logViewUpdateEnd(view: self)
-        let viewUpdateEndDate = Date()
+        let viewUpdateEndTime = CACurrentMediaTime()
```
Comment (kyleve) [inline BlueprintUI/Sources/BlueprintView/BlueprintView.swift:378]: Is this in seconds still? Fine if it's not, just curious.

Actually, why don't I just not be lazy and check, yep:

> Returns the current absolute time, in seconds.

**Review by kyleve — APPROVED** [review]

**Review by watt — APPROVED** [review]

File: BlueprintUI/Tests/BlueprintViewTests.swift (hunk 1 above)
Comment (robmaceachern) [reply BlueprintUI/Tests/BlueprintViewTests.swift]: Good call. An extra ulp never killed anyone 😛 59714ac

File: BlueprintUI/Sources/BlueprintView/BlueprintView.swift [hunk 3]
```diff
@@ -344,7 +344,7 @@ public final class BlueprintView: UIView {
             .layout(layoutAttributes: LayoutAttributes(frame: rootFrame), environment: environment)
             .resolve() ?? []
 
-        let measurementEndDate = Date()
+        let measurementEndTime = CACurrentMediaTime()
```
[the remark this answers is not in the document]
Comment (robmaceachern) [reply BlueprintUI/Sources/BlueprintView/BlueprintView.swift]: 👍🏽 59714ac

**Pull request merged**